### PR TITLE
OHRI-958 Refactor code for Encounter Tile Widget to make it more generic

### PR DESCRIPTION
**Pull request by kajambiya**

### CODE REVIEW

**Comment by kajambiya** [comment]
[](url)
Screenshot 2022-10-27 at 14 49 24

File: packages/esm-hiv-app/src/views/service-summary/encounter-list/service-summary-encounter-list.component.tsx [hunk 1]
```diff
@@ -46,29 +47,42 @@ const ServiceSummaryOverviewList: React.FC<OverviewListProps> = ({ patientUuid }
...
         encounterUuid: art_Therapy_EncounterUUID,
-        concept: regimen_UUID,
-        summaryConcept: regimenLine_UUID,
+        hasSummary: true,
+        getObsValue: (encounter) => {
+          return getObsFromEncounter(encounter, regimen_UUID);
+        },
+        getSummaryObsValue: (encounter) => {
+          return getObsFromEncounter(encounter, regimenLine_UUID);
+        },
```
Comment (pirupius) [inline packages/esm-hiv-app/src/views/service-summary/encounter-list/service-summary-encounter-list.component.tsx:64]: what's the difference between these 2?

File: packages/esm-commons-lib/src/components/encounter-tile/encounter-tile.component.tsx [hunk 2]
```diff
@@ -2,41 +2,23 @@ import { openmrsFetch } from '@openmrs/esm-framework';
...
-  isConceptDate?: Boolean;
-  isConceptSummaryDate?: Boolean;
-  isSummaryDaysCalculation?: Boolean;
-  obsValue?: string;
-  summaryConcept?: string;
-  summaryObsValue?: string;
-  isARTDateConcept?: Boolean;
+  getObsValue: (encounter: any) => string;
+  getSummaryObsValue?: (encounter: any) => string;
+  encounter?: any;
```
Comment (pirupius) [inline packages/esm-commons-lib/src/components/encounter-tile/encounter-tile.component.tsx:13]: do we need both `encounterUuid` and `encounterUuid`? what's the difference

Comment (kajambiya) [reply packages/esm-commons-lib/src/components/encounter-tile/encounter-tile.component.tsx:13]: I didn't get this. You wrote the same thing

**Review by larslemos — APPROVED** [review]

**Review by pirupius — APPROVED** [review]

**Pull request merged**